app/models/command.rb

```diff
@@ -27,4 +27,12 @@ def usages
   def self.usage_ids
     MacroCommand.pluck(:command_id) + StageCommand.pluck(:command_id)
   end
+
+  def to_auditable_json
+    to_json(
+      only: [:id, :command, :project_id],
+      include: [
+        stages: { only: [:id] }
+      ])
+  end
```

Comment:
How about we call the method `to_auditable_hash` and then use `as_json` instead of `to_json`.

As a rule of thumb, you should hardly ever need to call `to_json` except at the rendering step.  Models should return hashes, and then have something else be responsible for serializing the data.

Some more info: http://jonathanjulian.com/2010/04/rails-to_json-or-as_json/